src/utils/spark.py

```diff
@@ -231,15 +279,29 @@ def repartition(self) -> None:
             max_rows_per_file=5 * 10**6,
         )
 
-    def upload(self) -> None:
+        time_end = time.time()
+        time_duration = str(timedelta(seconds=(time_end - time_start)))
+        self.session.logger.info(
+            f"Table {self.table_name} repartitioned in {time_duration}"
+        )
+
+    def upload(self, aws: AWSClient) -> bool:
```

Comment:
The S3 client setup was originally in here, but it felt weird having a dedicated AWS class AND the client setup here, so I moved the connection the class.

Reply:
**[Praise]** Much cleaner separation of concerns!